preprocess_ecearth.py

```diff
@@ -102,46 +123,39 @@ def getWandFluxes(
 
     sp = np.exp(lnsp)  # log(sp) --> sp (Pa)
 
-    # Add surface and atmosphere together for u,q,v
-    u_top = u.isel(lev = -1)
-    u_top["lev"] = 10000
-    u_total = np.zeros((time, levelist.size + 2, len(latitude), len(longitude)))
-    u_total[:, 1:-1, :, :] = u
-    u_total[:, 0, :, :] = u10
-    u_total[:, -1, :, :] = u[:, -1, :, :]
-
-    v_total = np.zeros((time, levelist.size + 2, len(latitude), len(longitude)))
-    v_total[:, 1:-1, :, :] = v
-    v_total[:, 0, :, :] = v10
-    v_total[:, -1, :, :] = v[:, -1, :, :]
+    levelist = u.lev.values
+    time = q.shape[0]
 
-    q_total = np.zeros((time, levelist.size + 2, len(latitude), len(longitude)))
-    q_total[:, 1:-1, :, :] = q
-    q_total[:, 0, :, :] = q2m
+    u = join_levels(u, u10)  # bottom pressure will be set to 100000 hPa
+    v = join_levels(v, v10)
+    q = join_levels(q, q2m)
 
-    # Make pressure array
-    p = np.zeros((time, levelist.size + 2, len(latitude), len(longitude)))
-    p[:, 1:-1, :, :] = levelist[np.newaxis, :, np.newaxis, np.newaxis]
-    p[:, 0, :, :] = sp
+    u = repeat_top_level(u)  # top level will be set to 10000 hPa
+    v = repeat_top_level(v)
+    q = repeat_top_level(q)
```

Comment:
Note: in the original code zeros are inserted for q, instead of repeating the top level.